docs/non-normative/status-metrics.md

```diff
@@ -0,0 +1,95 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Status Metrics
+aliases: [status-metrics]
+--->
+
+# State Metrics
+
+**Status**: [Development][DocumentStatus]
+
+<!-- toc -->
+
+- [Definition](#definition)
+- [Design](#design)
+  - [Naming](#naming)
+  - [Instrument](#instrument)
+  - [Why not Resource Attributes?](#why-not-resource-attributes)
+
+<!-- tocstop -->
+
+## Definition
+
+For the purposes of this document, a "status metric" is a metric used to
+represent something being in a particular state at a given time from a closed
+set of distinct possible state values.
+
+## Design
+
+To represent a metric of this nature, the design will look like the following:
+
+```yaml
+id: metric.example.status
+type: metric
+metric_name: example.status
+stability: development
+brief: "The current state."
+note: |
+  A timeseries is produced for every possible value of `example.state`. The
+  value of this metric is 1 for a state if it is currently in said state, and
+  is 0 for all other states.
+instrument: updowncounter
+unit: "1"
+attributes:
+- ref: example.state
+```
+
+### Naming
+
+This pattern represents a unique naming challenge. You could reasonably call the
+metric and attribute both "status" or "state" as the two words are relatively
+interchangeable in common parlance.
+
+This problem necessitates choosing words to represent two things:
+
+* The noun representing one of the set of possible values (the attribute name)
+* The adjective describing which of the values is currently active (the metric
+  name)
+
+The general recommendation is to use the word "state" for the attribute and
+"status" for the metric. This is derived from common turns of phrase for each
+word, respectively:  
+"What **state** is X in?"  
+"What is the **current status** of X?"  
+
+The english-language semantics of this are heavily debatable, thus it is
+recommended for the sake of consistency that this pattern be adopted to avoid
+inconsistent naming across similarly designed metrics.
+
+The exception to this would be if a particular system has only one word that
+would be used for both the noun and the adjective in question. An example would
+be in Kubernetes, where the word "phase" equally represents both and there
+is no acceptable alternative. In this case, using a metric name suffix is
+recommended to avoid the naming clash, i.e. naming the attribute `k8s.phase` and
+the metric `k8s.phase.current`.
```

Comment:
Looking, I see there's some metrics where the name + attributes are the same, thus already going against the recommendation here. For ex:

https://github.com/open-telemetry/semantic-conventions/blob/83b55440146a1978699bc79b1c88b8a73085478c/model/k8s/metrics.yaml#L79, https://github.com/open-telemetry/semantic-conventions/blob/83b55440146a1978699bc79b1c88b8a73085478c/model/k8s/metrics.yaml#L92

Could those be somehow adapted to not clash like it is proposed here? Would be unfortunate that we publish the guidance while we itself already do not follow it, unless we can give reasons.

Reply:
I can loosen the guidance here. But isn't it pretty problematic to have an attribute that is the exact same name as a metric? They are un-clashed in tooling by their Weaver identifiers, but couldn't this cause problems or at least confusion for people interpreting the conventions?

Reply:
I was confused when reviewing some of the new k8s metrics that include a required attribute that exactly matches the metric name. I prefer the guidance suggested in the PR

Reply:
I have opened an issue to get clarification around this in general here: https://github.com/open-telemetry/semantic-conventions/issues/2476

Reply:
@TylerHelmuth we discussed this during the last System SemConv SIG (Sept 4th) with @open-telemetry/semconv-system-approvers and @jsuereth. The group in general is not convinced that this distinction should be enforced so as to support flattening fields which is not necessarily a pure OTel use-case/need. When transforming, I assume users can easily add a `metric.*` prefix to their metrics and/or `attribute.*` to the attributes to avoid such collisions.

Distinguishing them will make our lives hard on finding legit names for metrics like `k8s.container.status.reason` which are already long.

Reply:
No worries. In that case @braydonk do you think we can adjust the guidance accordingly? Or is there anything else missing to discuss here?